pkg/deployer/helm/ensure.go

```diff
@@ -74,40 +86,63 @@ func (h *Helm) ApplyFiles(ctx context.Context, files map[string]string, exports
 			},
 		}
 	}
+	h.ProviderStatus.ManagedResources = managedResources
 
-	if h.ProviderStatus != nil {
-		if err := h.cleanupOrphanedResources(ctx, targetClient, h.ProviderStatus.ManagedResources, objects); err != nil {
-			err = fmt.Errorf("unable to cleanup orphaned resources: %w", err)
-			return lsv1alpha1helper.NewWrappedError(err,
-				currOp, "CleanupOrphanedResources", err.Error())
-		}
+	if err := h.cleanupOrphanedResources(ctx, targetClient, h.ProviderStatus.ManagedResources, objects); err != nil {
+		err = fmt.Errorf("unable to cleanup orphaned resources: %w", err)
+		return lsv1alpha1helper.NewWrappedError(err,
+			currOp, "CleanupOrphanedResources", err.Error())
 	}
 
-	status := &helmv1alpha1.ProviderStatus{
-		TypeMeta: metav1.TypeMeta{
-			APIVersion: helmv1alpha1.SchemeGroupVersion.String(),
-			Kind:       "ProviderStatus",
-		},
-		ManagedResources: managedResources,
-	}
-	statusData, err := encodeStatus(status)
+	h.DeployItem.Status.ProviderStatus, err = encodeStatus(h.ProviderStatus)
 	if err != nil {
 		return lsv1alpha1helper.NewWrappedError(err,
-			currOp, "UpdateProviderStatus", err.Error())
+			currOp, "ProviderStatus", err.Error())
+	}
+	if err := h.kubeClient.Status().Update(ctx, h.DeployItem); err != nil {
+		return lsv1alpha1helper.NewWrappedError(err,
+			currOp, "UpdateStatus", err.Error())
 	}
 
 	if err := h.createOrUpdateExport(ctx, exports); err != nil {
 		return lsv1alpha1helper.NewWrappedError(err,
 			currOp, "CreateExport", err.Error())
 	}
```

Comment:
we should create the export after the resources are successfully reconciled.

Reply:
Fixed, creating the export is now very last step in the reconcile function.